### PR TITLE
Better global excludes

### PR DESCRIPTION
**Pull request by rsandor**
The way I was handling global excludes was incorrect. This changes the following:

* Moved rule commands into their own functions
* Instead of filtering the grep results, we now check each file (easiest way to exclude files and avoids argument lengths)

### CODE REVIEW

File: lib/script-generator.js [hunk 1]
```diff
@@ -195,12 +196,11 @@ ScriptGenerator.prototype.exclude = function (rule, index) {
     '# }'
   ].join('\n');
 
-  var excludes = rule.files.map(function (name) {
-    return '\\( ! -name \'' + name + '\' \\)';
-  }).join(' ');
+  var excludes = rule.files.filter(isString).map(function (name) {
+    return 'grep -v \'^./' + name.replace(/'/g, '') + '$\'';
+  }).join(' | ');
```
Comment (anandkumarpatel) [inline lib/script-generator.js]: this is kind of hard to follow. mind commenting it ?
what I see is it creates a string from a file array
the file array is first filtered to ensure it is a string
then with that list you generate a grep -v lines which excludes those files (removing `'` from the name)

File: lib/script-generator.js [hunk 2]
```diff
@@ -195,12 +196,11 @@ ScriptGenerator.prototype.exclude = function (rule, index) {
     '# }'
   ].join('\n');
 
-  var excludes = rule.files.map(function (name) {
-    return '\\( ! -name \'' + name + '\' \\)';
-  }).join(' ');
+  var excludes = rule.files.filter(isString).map(function (name) {
+    return 'grep -v \'^./' + name.replace(/'/g, '') + '$\'';
```
Comment (anandkumarpatel) [inline lib/script-generator.js]: should you replace `"` (double quote) as well

Comment (rsandor) [inline lib/script-generator.js]: Double quote will work fine since the names are single quoted.

File: lib/script-generator.js [hunk 3]
```diff
@@ -54,7 +55,7 @@ ScriptGenerator.prototype.preamble = function () {
 
   var common = [
     '_script_name=`basename $0`',
-    'search_files=\'.\'',
+    'search_files=`find . -type f | grep -v \'./.git\'`',
```
Comment (anandkumarpatel) [inline lib/script-generator.js]: this scares me a bit. bash has a argument limit (I think it was 5000) and if you have something like api (or even mavis) anytime you deference this and use it as input to something it will break.  the alternative is to convert this into a bash array, and loop where you use this

**Comment by bkendall** [comment]
went through a couple times, looks pretty good to me! :+1:
